### PR TITLE
Add additional CRD validation E2E tests

### PR DESCRIPTION
**Pull request by DangerOnTheRanger**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This PR adds additional E2E tests as part of the [test plan for KEP-2876](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/2876-crd-validation-expression-language#test-plan), with the long-term goal of getting the tests into conformance.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cici37** [comment]
The test got skipped in presubmit. I guess you have to use `framework.ConformanceIt` to add it into part of conformance test which will not be skipped.

**Comment by aojea** [comment]
> The test got skipped in presubmit. I guess you have to use `framework.ConformanceIt` to add it into part of conformance test which will not be skipped.

it doesn't work that way, conformance has some promotion rules, all test have to start as e2e tests before they move to conformance.

>var _ = SIGDescribe("CustomResourceValidationRules [Privileged:ClusterAdmin][Alpha][Feature:CustomResourceValidationExpressions]", func() {

The test are not running in presubmit because they are tagged as Alpha, is this an alpha feature, right?

/test

File: test/e2e/apimachinery/crd_validation_rules.go [hunk 1]
```diff
@@ -166,4 +166,59 @@ var _ = SIGDescribe("CustomResourceValidationRules [Privileged:ClusterAdmin][Alp
...
+			    "properties":{
+				  "x":{
+				    "type":"array",
+				    "items":{
+				      "type":"array",
+					  "items":{
+					    "type":"string"
+					  },
+					  "x-kubernetes-validations":[
+					    { "rule":"self.all(s, s == \"string constant\")" }
```
[the remark this answers is not in the document]
Comment (DangerOnTheRanger) [reply test/e2e/apimachinery/crd_validation_rules.go]: Fixed

**Comment by cici37** [comment]
/lgtm
/assign @liggitt 
Would you mind to take a look at the added e2e test for CRD expression validation when you have time? Thank you

**Pull request merged**